VerbrauchZeitspanne/module.php

```diff
@@ -18,19 +23,34 @@ public function ApplyChanges()
             //Never delete this line!
             parent::ApplyChanges();
 
+            //Get profile
+            $timeProfile = '';
+            $lod = $this->ReadPropertyInteger('LevelOfDetail');
+            switch ($lod) {
+                case 0:
+                    $timeProfile = '~UnixTimestampDate';
+                    break;
+                case 1:
+                    $timeProfile = '~UnixTimestampTime';
+                    break;
+                case 2:
+                    $timeProfile = '~UnixTimestamp';
+                    break;
+            }
+
             //Create variables
-            $this->RegisterVariableInteger('StartDate', 'Start-Datum', '~UnixTimestampDate', 1);
+            $this->RegisterVariableInteger('StartDate', 'Start-Datum', $timeProfile, 1);
             $this->EnableAction('StartDate');
 
-            if (GetValue($this->GetIDForIdent('StartDate')) == 0) {
-                SetValue($this->GetIDForIdent('StartDate'), time());
+            if (GetValue($this->GetIDForIdent('StartDate')) == 0 || $lod == 1) {
```

Comment:
Magst du hier noch schreiben, warum du $lod == 1 hier auch prüfst? Was ist die Idee dahinter? Ist es intuitiv, dass es nur bei einem "Neustart" auf heute gesetzt wird und die restlichen Tage nicht? @DrNiels: Wie findest du das?

Reply:
Ich vermute, die Idee ist es sich hier an den aktuellen Tag anzupassen, da ja $lod=1 Uhrzeit bedeutet. Das sollte meiner Meinung nach aber nicht hier passieren, sondern in Calculate entsprechend extrahiert werden. Also wenn die Start/EndTime auf 15:00 Uhr vor 3 Tagen steht, dann interessiert dich halt nur die 15:00 Uhr